packages/core/lib/src/core_widget_factory.dart

```diff
@@ -12,50 +12,48 @@ import 'core_html_widget.dart';
 
 /// A factory to build widgets.
 class WidgetFactory {
-  BuildOp _styleBgColor;
-  BuildOp _styleBlock;
-  BuildOp _styleBorder;
-  BuildOp _styleDisplayNone;
-  BuildOp _styleMargin;
-  BuildOp _stylePadding;
-  BuildOp _styleSizing;
-  BuildOp _styleTextDecoration;
-  BuildOp _styleVerticalAlign;
-  BuildOp _tagA;
-  BuildOp _tagBr;
-  BuildOp _tagFont;
-  BuildOp _tagHr;
-  BuildOp _tagImg;
-  BuildOp _tagPre;
-  BuildOp _tagQ;
-  TextStyleHtml Function(TextStyleHtml, String) __tsbFontSize;
-  TextStyleHtml Function(TextStyleHtml, String) _tsbLineHeight;
-  State _state;
-
-  HtmlWidget get _widget => _state?.widget;
+  BuildOp? _styleBgColor;
+  BuildOp? _styleBlock;
+  BuildOp? _styleBorder;
+  BuildOp? _styleDisplayNone;
+  BuildOp? _styleMargin;
+  BuildOp? _stylePadding;
+  BuildOp? _styleSizing;
+  BuildOp? _styleTextDecoration;
+  BuildOp? _styleVerticalAlign;
+  BuildOp? _tagA;
+  BuildOp? _tagBr;
+  BuildOp? _tagFont;
+  BuildOp? _tagHr;
+  BuildOp? _tagImg;
+  BuildOp? _tagPre;
+  BuildOp? _tagQ;
+  TextStyleHtml Function(TextStyleHtml, String)? __tsbFontSize;
+  TextStyleHtml Function(TextStyleHtml, String)? _tsbLineHeight;
+  State? _state;
+
+  HtmlWidget? get _widget => _state?.widget as HtmlWidget?;
 
   /// Builds [Align].
   Widget buildAlign(
-          BuildMetadata meta, Widget child, AlignmentGeometry alignment) =>
+          BuildMetadata meta, Widget child, AlignmentGeometry? alignment) =>
```

Comment:
I think for consistency, all `buildXxx` methods should return nullable type (either `Widget?` or `WidgetPlaceholder?`). Also, the `child` param and any additional param (e.g. `alignment` in this case) should be nullable too, unless it has some default value... Most apps will extend `WidgetFactory` so this class should be solid from the get go.

What do you think?

Reply:
That makes sense to me. I had to change some stuff to check for `null` return values.

I'm however not sure if it makes sense to make all params nullable, because the methods are called from this library and we never pass null for some arguments. If apps extend `WidgetFactory`, they would have to check for `null` for every argument, even if it can't actually be `null` because this library never passes null for this param.
For example `buildDecoratedBox` is only called with a `child` that is not null. Does it make sense to declare `Widget? child` as the parameter?
Maybe it makes sense to be consistent, but I don't know...

Reply:
I see your point. Then maybe we should make all the params not-nullable so WidgetFactory methods will never receive `null` as param + never return `null`? This change will require some big refactoring, let me know if you need my assistance on this.

Reply:
I think being able to _return_ `null` would still be useful to indicate that nothing should be rendered. An example would be an empty column/row.
I will try to have a closer look but can't really promise anything because I'll probably have less time this week :)

Reply:
> I think being able to return null would still be useful to indicate that nothing should be rendered.

This is true -> the return type should be nullable and all params should be non-nullable? That way all the complicated null check will be within the package 👍